test/factories/attachment_data.rb

```diff
@@ -7,7 +7,6 @@
 
       after(:build) do |attachment_data|
         attachment_data.assets << build(:asset, asset_manager_id: "asset_manager_id_original", variant: Asset.variants[:original], filename: attachment_data.filename)
-        attachment_data.assets << build(:asset, asset_manager_id: "asset_manager_id_thumbnail", variant: Asset.variants[:thumbnail], filename: "thumbnail_#{attachment_data.filename}.png")
```

Comment:
I recall that, at the time, we added a variety of factories to give us plenty to work with for both tests where "we only care if it has an asset, or not even an asset" and tests where "it would be really good to know that update fires twice". Now that we only have the one variant you could perhaps get rid of the doc factory and the `attachment_data_with_asset` and replace it with pdf. `csv` may have its own use cases so we should probably keep.

Reply:
Good point - thanks

Reply:
Had a quick look, definitely can remove the `attachment_data_with_asset` factory.

Reply:
Done!